lib/backend/syncersv1/updateprocessors/profileprocessor.go

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package updateprocessors
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/projectcalico/libcalico-go/lib/apiv2"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+	"github.com/projectcalico/libcalico-go/lib/backend/watchersyncer"
+	cnet "github.com/projectcalico/libcalico-go/lib/net"
+)
+
+// Create a new SyncerUpdateProcessor to sync Profile data in v1 format for
+// consumption by Felix.
+func NewProfileUpdateProcessor() watchersyncer.SyncerUpdateProcessor {
+	return &profileUpdateProcessor{
+		v2Kind: apiv2.KindProfile,
+	}
+}
+
+// Need to create custom logic for Profile since it breaks the values into 3 separate KV Pairs:
+// Tags, Labels, and Rules.
+type profileUpdateProcessor struct {
+	v2Kind string
+}
+
+func (pup *profileUpdateProcessor) Process(kvp *model.KVPair) ([]*model.KVPair, error) {
+	// Check the v2 resource is the correct type.
```

Comment:
I think this processing could be simplified a little by doing more of it inline rather than splitting into separate key/value functions.  That split doesn't work so well for the profile since it is composed of three things - and passing round slices of different keys and values that then need to be joined together feels a little awkward.

Also, since we don't need the tags any more, i'd judo drop both the key and value completely.

Reply:
Ok, I was thinking that it might've been a little easier to read if I broke off the key components into their own similarly named helper functions, but I can see how slices or maps being returned can be detrimental for that readability and a little more of a pain to maintain.